### PR TITLE
feat: support for GraphQL integrations

### PR DESCRIPTION
**Pull request by RomainMuller**
Added support for the available GraphQL integrations in the tracer library:

- `99designs/gqlgen`
- `graph-gophers/graphql-go`
- `graphql-go/graphql`

Also - fixed a bug in the line offset normalization that could result in incorrectly labeling modified file lines when a single-statement block was written as a one-liner; due to incorrectly assigning synthetic spacing at the start of it.

### CODE REVIEW

File: internal/injector/builtin/yaml/graphql/gqlgen.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    advice:
+      - wrap-expression:
+          imports:
+            handler: github.com/99designs/gqlgen/graphql/handler
+            tracer: gopkg.in/DataDog/dd-trace-go.v1/contrib/99designs/gqlgen
+          template: |-
+            func(s *handler.Server) *handler.Server {
+              s.Use(tracer.NewTracer())
+              return s
+            }({{ . }})
```
Comment (eliottness) [inline internal/injector/builtin/yaml/graphql/gqlgen.yml:27]: Optional
```suggestion
            gqlgentrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/99designs/gqlgen
          template: |-
            func(s *handler.Server) *handler.Server {
              s.Use(gqlgentrace.NewTracer())
              return s
            }({{ . }})
```

File: internal/injector/builtin/yaml/graphql/graph-gophers.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        - function-call: github.com/graph-gophers/graphql-go.MustParseSchema
+        - function-call: github.com/graph-gophers/graphql-go.ParseSchema
+    advice:
+      - append-args:
+          type: any
+          values:
+            - imports:
+                graphql: github.com/graph-gophers/graphql-go
+                trace: gopkg.in/DataDog/dd-trace-go.v1/contrib/graph-gophers/graphql-go
+              template: graphql.Tracer(trace.NewTracer())
```
Comment (eliottness) [inline internal/injector/builtin/yaml/graphql/graph-gophers.yml:26]:
```suggestion
                graphqltrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/graph-gophers/graphql-go
              template: graphql.Tracer(graphqltrace.NewTracer())
```
optional

File: samples/server/graphql.go
Comment (eliottness) [inline samples/server/graphql.go:1]: Optional: rename this file (and the .snap) to `gqlgen.go` if you don't want to do samples for the 3 of them

File: internal/injector/aspect/advice/call.go [hunk 3]
```diff
@@ -108,7 +108,9 @@ func (a *appendArgs) AsCode() jen.Code {
 
 func (a *appendArgs) AddedImports() []string {
 	imports := make([]string, 0, len(a.templates)+1)
-	imports = append(imports, a.typeName.ImportPath())
+	if argTypeImportPath := a.typeName.ImportPath(); argTypeImportPath != "" {
+		imports = append(imports, argTypeImportPath)
+	}
```
Comment (eliottness) [inline internal/injector/aspect/advice/call.go:113]: Codecov says that this is not covered. Can you either write a test or a comment?

Comment (RomainMuller) [reply internal/injector/aspect/advice/call.go:113]: That's odd because I added this code path to fix a bug... need to check this one!

Comment (RomainMuller) [reply internal/injector/aspect/advice/call.go]: I added a specific unit test for that ^^

**Review by eliottness — APPROVED** [review]

**Pull request merged**